scrapy_zyte_api/providers.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Callable, Sequence, Set
+
+from scrapy import Request
+from scrapy.crawler import Crawler
+from scrapy.utils.defer import maybe_deferred_to_future
+from scrapy_poet import PageObjectInputProvider
+from web_poet import BrowserResponse
+from zyte_common_items import Product
+
+from scrapy_zyte_api.responses import ZyteAPITextResponse
+
+try:
+    # requires Scrapy >= 2.8
+    from scrapy.http.request import NO_CALLBACK
+except ImportError:
+    NO_CALLBACK = None
+
+
+class ZyteApiProvider(PageObjectInputProvider):
+    name = "zyte_api"
+
+    provided_classes = {BrowserResponse, Product}
+
+    async def __call__(
+        self, to_provide: Set[Callable], request: Request, crawler: Crawler
+    ) -> Sequence[Any]:
+        """Makes a Zyte API request to provide BrowserResponse and/or item dependencies."""
+
+        # TODO what if ``response`` is already from Zyte API and contains something we need
+        zyte_api_meta = {}
+        if BrowserResponse in to_provide:
+            zyte_api_meta["browserHtml"] = True
+        if Product in to_provide:
+            zyte_api_meta["product"] = True
+        request = Request(
+            url=request.url,
+            meta={
+                "zyte_api": zyte_api_meta,
+            },
```

Comment:
What do we do about `ZYTE_API_DEFAULT_PARAMS`? Should we define a meta key to prevent them from being used? (e.g. `"zyte_api_default_params": False`)? Do we leave it for after merging this, and for now document that `ZYTE_API_DEFAULT_PARAMS` should not be used on spiders that use web-poet to get BrowserHtml, BrowserResponse or zyte_common_items.Product?

Reply:
I'm fine with both options.

Reply:
I've added `zyte_api_default_params` that makes it ignore all 3 settings (ZYTE_API_DEFAULT_PARAMS, ZYTE_API_AUTOMAP_PARAMS, ZYTE_API_TRANSPARENT_MODE).